.travis.yml

```diff
@@ -36,101 +36,101 @@ install: true
 
 jobs:
   include:
-    - stage: "Verify vendor"
-      script: make verify
-      name: "check if the vendor is up to date"
-
-    - stage: "Test lint"
-      script: make cloud_lint
-      name: "cloud_lint"
-    - script: make bluetoothdevice_lint
-      name: "bluetooth_device_lint"
-    - script: make keadm_lint
-      name: "keadm_lint"
-    - script: make edge_lint
-      name: "edge_lint"
-
-    - stage: "Test Build on amd64"
-      script: make
+    # Multiple stages in travis cannot run in parallel
+    # But jobs in a stage can run in parallel
+    # Now travis can only run 3 jobs in parallel
+    # So we just use 2 stages and 3 jobs in each stage
+    # The first stage is "ci on amd64", run in amd64 platform
+    # The second stage is "ci on arm64", run in arm64 platform
+    # And we put the longest-consuming job first
+    - stage: "ci on amd64"
+      name: "e2e test"
       arch: amd64
-      name: "builds cloud and edge components"
-    - script: make bluetoothdevice
+      script: 
+        # Now e2e_test takes the longest time, so we put it first
+        - make e2e_test 
+    
+    - name: "Verify Vendor, lint, build, crossbuild"
       arch: amd64
-      name: "build bluetooth_device"
-    - script: make edge_small_build
+      script: 
+        # check if the vendor is up to date
+        - make verify 
+        # cloud_lint
+        - make cloud_lint 
+        # bluetooth_device_lint
+        - make bluetoothdevice_lint 
+        # keadm_lint
+        - make keadm_lint 
+        # edge_lint
+        - make edge_lint 
+
+        # make cloudcore edgecore edgesite keadm
+        - make 
+        # build bluetooth_device
+        - make bluetoothdevice 
+        # build small edge version
+        - make edge_small_build 
+
+        # cross build edge
+        - make edge_cross_build 
+        # cross build edge for armv7
+        - make edge_cross_build_v7 
+        # cross build edgesite
+        - make edgesite_cross_build 
+        # cross build edgeite for armv7
+        - make edgesite_cross_build_v7 
+
+    - name: "build docker image, unit test and integration test" 
       arch: amd64
```

Comment:
If ut or integration test failed, we will need to rerun the build docker image here? I think we can put all the build and build image in a job, thought？

Reply:
> If ut or integration test failed, we will need to rerun the build docker image here?
yes

> I think we can put all the build and build image in a job, thought？
I did this at the beginning, but this will be turned into 4 jobs, because only 3 jobs can be executed concurrently at a time, the 4th job must wait until the first 3 jobs are completed, so the total time Becomes very long(Virtual machine creation also takes time). In addition, the build time of the image is relatively short, so this does not affect

Reply:
done